src/core/services/user.py

```diff
@@ -58,3 +58,17 @@ async def set_mailing(self, telegram_id: int) -> bool:
             user = await repository.get_by_telegram_id(telegram_id)
             await repository.set_mailing(user, not user.has_mailing)
             return user.has_mailing
+
+    async def check_and_set_has_mailing_atribute(self, telegram_id: int) -> None:
+        """
+        Присваивает пользователю атрибут has_mailing, для получения почтовой
+        рассылки на задания после выбора категорий. Предварительно
+        осуществляется проверка, установлен ли этот атрибут у пользователя
+        ранее.
+        """
+        async with self._sessionmaker() as session:
+            repository = UserRepository(session)
+            user = await repository.get_by_telegram_id(telegram_id)
+            has_mailing = user.has_mailing
+            if not has_mailing:

```

Comment:
Старайся использовать меньше переменных, так будет читабельнее:
```suggestion
            if not user.has_mailing:
```